filebeat/input/log/input.go

```diff
@@ -135,6 +116,25 @@ func NewInput(
 		return nil, err
 	}
 
+	// Note: underlying output.
+	//  The input and harvester do have different requirements
+	//  on the timings the outlets must be closed/unblocked.
+	//  The outlet generated here is the underlying outlet, only closed
+	//  once all workers have been shut down.
+	//  For state updates and events, separate sub-outlets will be used.
+	out, err := outlet(cfg, context.DynamicFields)
+	if err != nil {
+		return nil, err
+	}
+
+	// stateOut will only be unblocked if the beat is shut down.
+	// otherwise it can block on a full publisher pipeline, so state updates
+	// can be forwarded correctly to the registrar.
+	stateOut := channel.CloseOnSignal(channel.SubOutlet(out), context.BeatDone)
+
+	p.outlet = out
+	p.stateOutlet = stateOut
```

Comment:
this can panic. The `loadStates` call checks each state and sends an update event to `p.outlet`. If there is some matching state in the registry you will have a nil pointer.

Reply:
Ok, I will go back to previous version.